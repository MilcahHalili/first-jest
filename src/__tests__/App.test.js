import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer';

describe('my first snapshot test', () => {
  test('testing app button', () => {
    const tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})

describe('changing button name to Hide', () =>{
  test('toggle the button', () => {
    const tree = create(<App />);
    const instance = tree.getInstance();
    expect(instance.state.isActive).toBe(false)
    // changing the state
    instance.handleClick();
    // isActive property is updated to true
    expect(instance.state.isActive).toBe(true);
    expect(tree.toJSON()).toMatchSnapshot()
  })
})