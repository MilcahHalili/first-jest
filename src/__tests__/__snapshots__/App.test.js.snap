// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changing button name to Hide toggle the button 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Show
  </button>
  <button
    onClick={[Function]}
  >
    SUBSCRIBE TO BASIC
  </button>
</div>
`;

exports[`my first snapshot test testing app button 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Show
  </button>
  <button
    onClick={[Function]}
  >
    SUBSCRIBE TO BASIC
  </button>
</div>
`;
